numpy/core/tests/test_umath.py

```diff
@@ -1918,20 +1920,27 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
 
                 results = super(A, self).__array_ufunc__(ufunc, method,
                                                          *args, **kwargs)
-                if not isinstance(results, tuple):
-                    if not isinstance(results, np.ndarray):
-                        return results
+                if results is NotImplemented:
+                    return NotImplemented
+
+                if ufunc.nout == 1:
```

Comment:
Does `ufunc.nout` always mean the right thing for `reduce`, `reduce_at`, `at` and `accumulate`?